StRoot/StarGenerator/Hijing1_383/StarHijing.cxx

```diff
@@ -15,6 +14,28 @@ using namespace std;
 
 #include "TGenericTable.h"
 
+#include "Hijing.h"
+
+namespace {
+  vector<std::string> tokenize(const std::string &s, const std::string &sep_chars=";")
+  {
+    std::string::size_type prev_pos = 0, pos = 0;
+    std::vector<std::string> result;
+    
+    if ( s.length() > 1 ) {
+      while ((pos = s.find_first_of(sep_chars, pos)) != std::string::npos) {
+	result.push_back(  ( s.substr(prev_pos, pos - prev_pos))  );
+	pos += 1;
+	prev_pos = pos;
+      }
+      result.push_back( (s.substr(prev_pos)));
+    }
+
+    return result;
+  }
+
+}
```

Comment:
An alternative coding I saw which looked rather elegant to me, but requires `#include <sstream>`. I'm not sure if you'd still need the `if ( s.length() > 1 )` condition?
```suggestion
 std::vector<std::string> tokenize(const std::string& s, char delimiter = ';') {
    std::vector<std::string> result;
    std::istringstream iss(s); // Create a string stream from the input string
    std::string token;

    while (std::getline(iss, token, delimiter)) result.push_back(token);

    return result;
}```